### DISCUSSION TITLE
Multiple idle database connections

### DISCUSSION THREAD

**Discussion by danieljwestman** [directus/directus#3999] [Archive]
Directus v9.0.0-rc.22 (npx running on DO App Platform)
PG 12 (managed db at DO)
--

I just noticed that Directus is opening up 5-7 db connections when refreshing the browser. Most of them directly becomes <idle> in state, but they still exist. With 3-5 apps running against the same db cluster, not the same database, the maximum connections is quickly reached and you're on lock-down... 

Shouldn't unused connections be closed by default? Or am I getting this wrong?

I get a lot of activities (connections) running `SELECT * FROM pg_stat_activity`

Error in console:

![log](https://user-images.githubusercontent.com/20950568/101831375-19ad6980-3b36-11eb-9571-dd09e6d3dcd8.png)

**Comment by rijkvanzanten** [comment]
> Shouldn't unused connections be closed by default? Or am I getting this wrong?

Directus keeps multiple connections open at a time to increase performance. This is by design.

I'm down to discuss a way to explicitly configure the connection pooling settings of Directus 👍

**Comment by danieljwestman** [comment]
For general information; If you're running Directus (or another app) on DigitalOceans App Platform and use their Managed Databases service as the database component you are not able to use private networking. The connection must be public and therefor outsiders can try to connect. These connection attempts (hopefully failed) are consuming the connections limit which, obviously, can lead to problems for your apps (not being able to connect to the db due to limit breach (pretty low limits on the basic db options at DO)). 

I had a lot of unknown connection attempts to my, newly created, app/db at DO and asked the support about it. Comment from DO support team:

> So currently with App Platform your database connection must be publicly accessible. As such it is possible that outsiders may try to connect to your database. This is most likely what those connections are. This is definitely not the best option for database connections and our engineers are working on using private connections between your database and the app. This is aways off though and will take a bit of time to complete. 

This is not an issue regarding Directus per se, but I wanted to give a heads-up for other users.

**Reply by BlackDahlia313** [reply]
Just now running into this. Thank you for the insight. 

This was eventually added for anyone who is seeing this. You can now select an app in the trusted resource list for a managed database.
